Add release notes for 2.10.12 (#9812)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes
     
## Release 2.10.12

> 

Reviewer: ```suggestion

### The return of the deployments page
We got a lot of positive feedback about the new flows page that was redesigned to include deployments, but several users pointed out that the it wasn't quite a full replacement for the dedicated deployments page. The deployments page has been re-added to the navigation menu until the new flows page is a worthy substitute.

See the [pull request](https://github.com/PrefectHQ/prefect/pull/9800) for more information.

```

### Enhancements

> - Duplicate schemas client side and remove server imports to improve CLI start time — https://github.com/PrefectHQ/prefect/pull/9577

Reviewer: ```suggestion
- improve import times by duplicating schemas client side and remove server imports to improve CLI start time — https://github.com/PrefectHQ/prefect/pull/9577
```

- Add task run as a related object to emitted events — https://github.com/PrefectHQ/prefect/pull/9759
- Emit task run state change events when orchestrating a task run — https://github.com/PrefectHQ/prefect/pull/9684
- Add healthcheck webserver to workers — https://github.com/PrefectHQ/prefect/pull/9687
- Create files and directories with user-scoped permissions — https://github.com/PrefectHQ/prefect/pull/9789

> - Restore "Deployments" page in UI — https://github.com/PrefectHQ/prefect/pull/9800

Reviewer: ```suggestion
```
